### DISCUSSION TITLE
I for some reason cannot connect my github pages and squarespace custom domain. NotServedByPagesError

### DISCUSSION THREAD

**Discussion by ChristianC561** [community/community#133994] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have done everything it said to do and I still cannot figure out what I missed. I had it working awhile ago and then I changed my website. Now I am trying to just use my old one and it seems to not want to work. 

https://github.com/ChristianC561/Greenview

My DNS settings look like this:

greenviewsearch.com | A | N/A | 185.199.108.153
greenviewsearch.com | A | N/A | 185.199.109.153
greenviewsearch.com | A | N/A | 185.199.110.153
greenviewsearch.com | A | N/A | 185.199.111.153
www | CNAME | N/A | christianc561.github.io

I have the CNAME file in main also with just the URL inside. 

DNS check is giving me NotServedByPagesError

**Comment by LiteBrite82** [comment]
Hi @ChristianC561!

Thanks for posting!!!

I have a couple of questions
- Are you using an actions workflow to deploy your web page? 
- Did you add your custom domain name in the pages settings?
![Screenshot 2024-07-29 at 2 44 37 PM](https://github.com/user-attachments/assets/8963d709-fb89-409a-9c1c-1c5fa36ba6ca)

**Reply by ChristianC561** [reply]
Hi @LiteBrite82 firstly, thank you so much for commenting on my post!!!!!

Secondly, I did here is a screenshot. Let me know what you think. o7
![Screenshot (43)](https://github.com/user-attachments/assets/6b6082e3-78e8-4d68-a89f-afbefc4bf801)

**Reply by LiteBrite82** [reply]
Double check the steps in [this video](https://www.youtube.com/watch?v=EX4w9hsduNA) from @tonyflo so we know we have all of the basic bases covered.

**Reply by LiteBrite82** [reply]
Also,  just for the fun of it, create an empty file named `.nojekyll` in the root of your main branch. 
- Pages recently changed the way deployments are processed; its hard to explain. I have found, when in doubt, add a `.nojekyll' file. (unless you are in fact using jekyll to build your page).

**Reply by LiteBrite82** [reply]
(re run deployment after adding the `.nojekyll` file)

**Reply by LiteBrite82** [reply]
You could also try out the newer GitHub Actions Workflow deployment option. Alex from Ideaspot (@Ideaspotclass) has a great [video tutorial](https://www.youtube.com/watch?v=e5AwNU3Y2es) for that one. 

In this case you would remove the CNAME file you created but keep the `.nojekyll` file I recommended.

**Reply by ChristianC561** [reply]
> Double check the steps in [this video](https://www.youtube.com/watch?v=EX4w9hsduNA) from @tonyflo so we know we have all of the basic bases covered.

So the only thing I noticed that was different is he put all the a records for the IP addresses in the same record, while mine are in 4 separate records. Would that matter?

**Reply by ChristianC561** [reply]
Also this is on squarespace

**Reply by LiteBrite82** [reply]
> So the only thing I noticed that was different is he put all the a records for the IP addresses in the same record, while mine are in 4 separate records. Would that matter?

Can you post a screenshot?

**Comment by LiteBrite82** [comment]
@ChristianC561 

I dug a little deeper with one of our Pages Devs. From what we can tell, the DNS settings you are using have not updated yet. 

> greenviewsearch.com | A | N/A | 185.199.108.153
greenviewsearch.com | A | N/A | 185.199.109.153
greenviewsearch.com | A | N/A | 185.199.110.153
greenviewsearch.com | A | N/A | 185.199.111.153
www | CNAME | N/A | christianc561.github.io

It can take up to 24 hours to update

**Reply by LiteBrite82** [reply]
<img width="983" alt="Screenshot 2024-07-29 at 7 45 25 PM" src="https://github.com/user-attachments/assets/5722d892-9725-44c5-a15a-1607b07b42ca">

**Comment by LiteBrite82** [comment, marked as answer]
@ChristianC561 

We found that the `WWW` variant of your DNS is running (www.greenviewsearch.com) NOT greenviewsearch.com - Thats on the SquareSpace side. 

<img width="548" alt="Screenshot 2024-07-29 at 7 47 15 PM" src="https://github.com/user-attachments/assets/aedcc2d3-2489-4c19-825e-df09dbefb855">

What you can do, for now, is change the "Custom Domain" name in the Pages Settings to `www.greenviewsearch.com`.
You can change it back to `greenviewsearch.com` when the DNS finally updates.

**Reply by LiteBrite82** [reply]
I just found this on the squarespace website: [Removing the "www" from your domain](https://support.squarespace.com/hc/en-us/articles/206541837-Removing-the-www-from-your-domain)

This might actually be the source of the problem...

**Reply by LiteBrite82** [reply]
SWEET!!!!! https://www.greenviewsearch.com 

![Screenshot 2024-07-29 at 8 13 42 PM](https://github.com/user-attachments/assets/22b48f05-366d-437c-abc8-a327d7afce28)

**Reply by LiteBrite82** [reply]
That www thing seems so backward to me...... why does SquareSpace do that?

**Reply by ChristianC561** [reply]
> That www thing seems so backward to me...... why does SquareSpace do that?

Oh my gosh thank you so much!!!!!! Yeah it was ever since all the google domains went to Squarespace something changed. Man, I will tell you I thought I was going crazy! Such a simple fix lol thanks again you are the best

**Reply by LiteBrite82** [reply]
We just solved so much pain for so many others; Im making this the default thread for squarespace DNS issues.

**Reply by LiteBrite82** [reply]
**SYNOPSYS:**
SquareSpace uses the `www` variant by default, and redirects all "naked domain" requests to `www`. 

SO: `example.com` gets redirected to `www.example.com`; which is kinda counter to every tutorial I have ever seen. 

**GitHub Solution for SquareSpace NotServedByPagesError:**
If you are using SquareSpace for DNS Hosting, in GitHub Pages Settings for Custom Domain Name, be sure to preface your Domain Name with `www`

**Correct for SquareSpace:**
![Screenshot 2024-07-29 at 8 42 41 PM](https://github.com/user-attachments/assets/19e5c8d1-88b5-4901-969d-f1ae86259d50)

**Incorrect for SquareSpace:**
![Screenshot 2024-07-29 at 8 43 32 PM](https://github.com/user-attachments/assets/1f0a619c-2dda-4c56-aff2-62a3ad3ecf96)

**Reply by ChristianC561** [reply]
> We just solved so much pain for so many others; Im making this the default thread for squarespace DNS issues.

Glad we could figure this out, and hopefully help other people that will inevitably have the same issue

**Reply by TheYearlock** [reply]
i love you @LiteBrite82

**Comment by gitgurugroove** [comment]
It sounds incredibly frustrating when something that was working suddenly stops, especially after a change like a website migration. Let's break down what might be happening with your GitHub Pages site for greenviewsearch.com and the "NotServedByPagesError."

Based on your description and the provided DNS settings, here's where the most likely issues lie:

Incorrect A Records for GitHub Pages:
The A records you have configured (185.199.108.153, 185.199.109.153, 185.199.110.153, 185.199.111.153) are not the current, official IP addresses for GitHub Pages custom domains. GitHub periodically updates these IPs. If you had it working "awhile ago," these might have been the correct ones then, but they are no longer.

Solution: You need to update your A records to the current GitHub Pages IP addresses. As of my last update, these are:

185.199.108.153

185.199.109.153

185.199.110.153

185.199.111.153

Wait, hold on a second! I've just cross-referenced your provided IP addresses with the current GitHub Pages IP addresses. They are the same. This means the A records themselves are likely not the primary issue. This suggests the problem lies elsewhere.

Let's re-evaluate based on the NotServedByPagesError and your current setup.

The NotServedByPagesError typically means that GitHub's servers aren't seeing your domain configured correctly on their end, or there's a mismatch.

Here's a more targeted troubleshooting plan:

Verify Your Repository's GitHub Pages Settings:

Go to your repository: https://github.com/ChristianC561/Greenview

Click on Settings (top right, near "Code" and "Actions").

In the left sidebar, click on Pages.

Under "Custom domain," ensure that greenviewsearch.com is entered correctly.

Crucially: After entering the domain, wait for GitHub to perform its check. It should show a green checkmark indicating "Your custom domain is set up." If it shows an error or "unavailable," that's where the problem is originating.

Enforce HTTPS: Ensure the "Enforce HTTPS" checkbox is checked. This is highly recommended and can sometimes influence how the domain is served.

The CNAME File in Your Repository:
You mentioned you have a CNAME file in main with just the URL inside.

Content: The CNAME file in your repository's root (the main branch) should contain only your custom domain, like this:

greenviewsearch.com
Make sure there are no extra spaces, newlines, or other characters.

Location: It must be in the root of your main branch (or whichever branch you've configured GitHub Pages to serve from, usually main or gh-pages).

Case Sensitivity: Ensure the file name is exactly CNAME (uppercase).

DNS Propagation Time:
Even if you've done everything correctly, DNS changes can take time to propagate across the internet. While usually minutes to a few hours, in rare cases, it can take up to 48 hours. If you just made changes, give it some time. You can use a tool like [whatsmydns.net](https://www.whatsmydns.net/) to check if your A records and CNAME for www are resolving correctly globally.

Confirm the GitHub Pages Source:
On the Pages settings page in your repository, also confirm under "Build and deployment" that the "Source" is set correctly (e.g., "Deploy from a branch" and then select main and /root).

Check for Conflicting CNAME Records (Less Likely for NotServedByPagesError but worth a look):
While your www CNAME is correctly pointing to christianc561.github.io, ensure there isn't another conflicting CNAME record for greenviewsearch.com itself (the root domain). Only A records should be for the root domain.

Summary of what to check, in order of likelihood:

GitHub Pages Settings in your repository: Is greenviewsearch.com entered under "Custom domain" and is GitHub showing it as configured correctly there? This is the most critical step.

CNAME file: Is it in the root of your main branch, named exactly CNAME, and containing only greenviewsearch.com?

DNS Records (Confirm again): While the IPs look correct, ensure they are exactly 185.199.108.153, 185.199.109.153, 185.199.110.153, 185.199.111.153 for your bare domain (greenviewsearch.com), and your www CNAME points to christianc561.github.io.

Clear your browser cache/try incognito: Sometimes old redirects or cached DNS lookups can cause issues on your local machine.

If you go through these steps methodically, you should be able to pinpoint where the disconnection is happening. The "NotServedByPagesError" nearly always points to a misconfiguration on the GitHub repository's Pages settings, or an incorrect CNAME file. Good luck!
